test/core/util/BUILD

```diff
@@ -227,3 +227,22 @@ grpc_cc_library(
     srcs = ["build.cc"],
     hdrs = ["build.h"],
 )
+
+grpc_cc_library(
+    name = "gtest_main",
+    srcs = ["gtest_main.cc"],
+    external_deps = ["gtest"],
+    language = "C++",
+    deps = [
+        ":grpc_test_util_base",
```

Comment:
This pulls in all of //:grpc which I think is the wrong default, especially moving forward.

I'd propose a gtest_main that is free of that dependency for unit tests, and a gtest_main_for_end2end that does include the extras.